core/domain/exp_domain.py

```diff
@@ -2331,6 +2331,35 @@ def _convert_states_v29_dict_to_v30_dict(cls, states_dict):
                 del answer_group['tagged_misconception_id']
         return states_dict
 
+    @classmethod
+    def _convert_states_v30_dict_to_v31_dict(cls, states_dict):
+        """Converts from version 30 to 31. Version 31 updates the
+        Voiceover model to have an initialized duration attribute of 0.0.
+        This will be updated when a new mp3 audio file is uploaded
+        for the exploration.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for state_dict in states_dict.values():
+            # Get the voiceovers_mapping metadata.
+            voiceovers = (state_dict['recorded_voiceovers']
+                          ['voiceovers_mapping'])
+            for content_id in voiceovers.keys():
+                for language_code in voiceovers[content_id].keys():
+                    # Initialize duration with 0.0 for every voiceover
+                    # recording under Content, Feedback, Hints, Solutions.
+                    # This is necessary to keep the state functional
+                    # when migrating to v31.
+                    voiceovers[content_id][language_code]['duration'] = 0.0
```

Comment:
I see 0.0 here but  0 in the draft_upgarde service. should we also make that 0.0?

Reply:
yes that should be the case.

Reply:
Done.